### PR TITLE
CI: validate JSON & fix benchmark

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding validation for action YAMLs and fix them

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -64,6 +64,16 @@ repos:
...
 
+  - repo: https://github.com/sirosen/check-jsonschema
+    rev: 0.4.0
+    hooks:
+      - id: check-github-workflows # Validate GitHub Workflows against the schema provided by Schemastore
+        files: ^\.github/workflows/
+        types: [ yaml ]
+      - id: check-azure-pipelines # Validate Azure Pipelines config against the schema provided by Microsoft
+        files: ^\.azure-pipelines/
+        types: [ yaml ]
```
Comment (justusschock) [inline .pre-commit-config.yaml:75]: Is this really something that should be checked at every commit? That's the original purpose of pre-commit and many people (including me) use it that way. So I want it to be as lightweight as possible.

Comment (Borda) [reply .pre-commit-config.yaml:75]: well it shall be fast, on the other hand, you can say that you do not want to check python code when you do only edits in docs... :rabbit: but yes agree that it shall be light :)

Comment (justusschock) [reply .pre-commit-config.yaml:75]: I'm fine with docs. But I don't think we need to verify our CI-Pipelines every time :)

**Review by justusschock — APPROVED** [review]

File: .azure-pipelines/gpu_benchmark.yml [hunk 2]
```diff
@@ -5,17 +5,13 @@ on:
...
+    timeoutInMinutes: 90
     cancelTimeoutInMinutes: 2
-
     pool: gridai-spot-pool
-
     container:
       # base ML image: mcr.microsoft.com/azureml/openmpi3.1.2-cuda10.2-cudnn8-ubuntu18.04
-      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.8-torch1.6"
-
+      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.8-torch1.7"
```
Comment (carmocca) [inline .azure-pipelines/gpu_benchmark.yml:14]: should this use 1.8 as the gpu ci image?

**Review by carmocca — APPROVED** [review]

**Comment by Borda** [comment]
need to be resolved https://github.com/microsoft/azure-pipelines-vscode/issues/411

**Review by awaelchli — APPROVED** [review]

**Pull request merged**